### PR TITLE
New project

### PR DESCRIPTION
**Pull request by erpas**

### CODE REVIEW

File: Mergin/create_project_wizard.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    """Initial wizard page with Mergin project sources to choose."""
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setupUi(self)
+        self.parent = parent
+        self.btns_page = {
+            self.basic_proj_btn: SAVE_PAGE,
+            self.cur_proj_no_pack_btn: LOCATE_PAGE,
+            # self.cur_proj_pack_btn: SAVE_PAGE,
```
Comment (DamnBack) [inline Mergin/create_project_wizard.py:34]: Shouldn't it be removed?

File: Mergin/create_project_wizard.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+
+class ChoosePathPage(uicls_local_path_page, basecls_local_path_page):
+    """Page for getting local path for saving new Mergin project."""
+
+    def __init__(self, question=None, existing=None, file_filter=None, parent=None):
+        super().__init__(parent)
+        self.setupUi(self)
+        self.parent = parent
+        self.file_filter = "QGIS projects (*.qgz *.qgs)" if file_filter is None else file_filter
```
Comment (DamnBack) [inline Mergin/create_project_wizard.py:66]: QGIS open project have also *.QGZ *.QGS listed in filters. Could you add them as well?

File: Mergin/create_project_wizard.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        self.warning_edit.document().setPlainText("Checking the directory...")
+        QApplication.setOverrideCursor(Qt.WaitCursor)
+        qgis_files = find_qgis_files(cur_dir)
+        QApplication.processEvents()
+        QApplication.restoreOverrideCursor()
+
+        qgis_files_nr = len(qgis_files)
+        if self.file_path not in qgis_files:
+            qgis_files_nr += 1
+        if len(cur_dir) > 5 and ".mergin" in os.listdir(cur_dir):
```
Comment (DamnBack) [inline Mergin/create_project_wizard.py:143]: It would be nice to define variable for value '5'.

File: Mergin/projects_manager.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+            project_name = mp.metadata["name"]
+        except InvalidProject as e:
+            msg = f"Failed to get project status:\n\n{str(e)}"
+            QMessageBox.critical(None, "Project status", msg, QMessageBox.Close)
+            return
+        validator = MerginProjectValidator(mp, self.mc)
+        validation_results = validator.run_checks()
+        try:
+            pull_changes, push_changes, push_changes_summary = self.mc.project_status(project_dir)
+            # if not sum(len(v) for v in list(pull_changes.values()) + list(push_changes.values())):
```
Comment (DamnBack) [inline Mergin/projects_manager.py:156]: Please remove commented code.

File: Mergin/utils.py [hunk 5]
```diff
@@ -212,19 +223,217 @@ def validate_mergin_url(url):
...
+        if proj_path:
+            if not os.path.exists(proj_path):
+                proj_path = None
+        return proj_path
+
+    qgis_project_path = QgsProject.instance().absolutePath()
+    if not qgis_project_path:
+        return None
+
+    for local_path, _, _ in get_local_mergin_projects_info():
```
Comment (DamnBack) [inline Mergin/utils.py:435]: Would't it better to use meaningful variable names?

**Comment by erpas** [comment]
Some screenshots with the new wizard for Mergin project creation:

2021-02-01 17_50_05-VirtualBoxVM


2021-02-01 17_50_35-VirtualBoxVM


2021-02-01 17_51_04-VirtualBoxVM


2021-02-01 17_51_19-VirtualBoxVM

**Comment by erpas** [comment]
Some screenshots

new_proj_wiz_01

new_proj_wiz_02

new_proj_wiz_03

new_proj_wiz_04

**Comment by DamnBack** [comment]
Code looks great and new wizard works smoothly. @wonder-sk Feel free to have a second look. :)

**Comment by saberraz** [comment]
For New Basic QGIS project, it is still a bit confusing. User needs to create an empty folder and write the project name. Is it possible to simplify it and use + button to create a folder. The project name can be same as project folder.

**Comment by saberraz** [comment]
Consider removing the message at the bottom of this window:
image

When you can click **Next**, it means we are good to go.

**Comment by saberraz** [comment]
In project created from wizard for New Basic project, the **fid** field should be hidden.

**Comment by saberraz** [comment]
Similar to https://github.com/lutraconsulting/qgis-mergin-plugin/pull/194#issuecomment-790514780 IMO, Package Current QGIS project should select a folder only and assign the same name as the folder to the project.

**Comment by saberraz** [comment]
This looks great @erpas. 

When I click Finish to create the project, the wizard window remains open:
image

It gets closed when I close "Create project" window. Not a big deal.

**Comment by saberraz** [comment]
An issue reported by user. Not sure because the profile has spaces (FDB test 2)?
image

**Comment by saberraz** [comment]
I thought we do the project  creation under the ../ previous project (one level higher). If we use the last path (i.e. the root of the previous project), users might not change the default folder and it will end up several mergin project within each other:
image

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**